File: app/src/main/java/com/hugh/memorandum/BaseActivity.java
package com.hugh.memorandum;

import android.app.Activity;
import android.os.Bundle;
import android.support.annotation.Nullable;

public abstract class BaseActivity extends Activity {

    public abstract int setContentLayout();
    public abstract void initView();
    public abstract void initData();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(setContentLayout());
        initView();
        initData();
    }
}

File: app/src/main/java/com/hugh/memorandum/activity/BaseActivity.kt
package com.hugh.memorandum.activity

import android.app.Activity
import android.os.Bundle

abstract class BaseActivity : Activity() {

    abstract fun setContentLayout(): Int
    abstract fun initView()
    abstract fun initData()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(setContentLayout())
        initView()
        initData()
    }
}
